cmd/nerdctl/volume_rm.go

```diff
@@ -39,14 +43,52 @@ func newVolumeRmCommand() *cobra.Command {
 }
 
 func volumeRmAction(cmd *cobra.Command, args []string) error {
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	containers, err := client.Containers(ctx)
+	if err != nil {
+		return err
+	}
+	var mount *specs.Spec
+	var removednames []string
 	volStore, err := getVolumeStore(cmd)
 	if err != nil {
 		return err
 	}
 	names := args
-	removedNames, err := volStore.Remove(names)
-	for _, removed := range removedNames {
-		fmt.Fprintln(cmd.OutOrStdout(), removed)
+
+	for _, name := range names {
+		var found bool
+		for _, container := range containers {
+			n, err := containerinspector.Inspect(ctx, container)
+			if err != nil {
+				return err
+			}
+			err = json.Unmarshal(n.Container.Spec.Value, &mount)
+			if err != nil {
+				return err
+			}
+			volume, _ := volStore.Get(name)
+			if found = checkVolume(&mount.Mounts, volume.Mountpoint); found {
+				found = true
+				logrus.WithError(fmt.Errorf("Volume %q is in use", name)).Error(fmt.Errorf("Remove Volume: %q failed", name))
+				break
+			}
+		}
+		if !found {
+			// volumes that are not mounted to any container
+			removednames = append(removednames, name)
+		}
+	}
+	if removednames != nil {
+		rNames, err := volStore.Remove(removednames)
+		if err != nil {
+			panic(err)
```

Comment:
I think just return error directly would be better

Reply:
Ill change it.